modules/telemetry/telemetry/examples/log_sink_example.cpp

```diff
@@ -0,0 +1,23 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+#include <memory>
+
+#include "hephaestus/telemetry/log.h"
+#include "hephaestus/telemetry/log_sink.h"
+#include "hephaestus/telemetry/log_sinks/absl_sink.h"
+#include "hephaestus/utils/stack_trace.h"
+
+// NOLINTNEXTLINE(bugprone-exception-escape)
+auto main(int /*argc*/, const char* /*argv*/[]) -> int {
+  const heph::utils::StackTrace stack;
+
+  heph::telemetry::registerLogSink(std::make_unique<heph::telemetry::AbslLogSink>());
+
+  heph::log(heph::LogLevel::WARN, "testing absl log sink");
+
+  const int num = 1234;
+  heph::log(heph::LogLevel::INFO, "absl log sink with fields", "num", num, "quoted_string", "test");
```

Comment:
great, thanks guys for adjusting, how about the fields being pairs, ie.e 
```suggestion
  heph::log(heph::LogLevel::INFO, "absl log sink with fields", {"num", num}, {"quoted_string", "test"});
```
makes it clearer to what belongs together in a key-value way, no?

Reply:
yes definitely, this was our prefered solution, however I struggled combining brace initializers with variadics...
see https://github.com/olympus-robotics/hephaestus/pull/211
what your take, would you prefer
```
heph::log(heph::LogLevel::INFO, "absl log sink with fields", {{"num", num}, {"quoted_string", "test"}});
```
? @filippobrizzi and I prefered the interface with less to write...

Reply:
thats ok, lets keep the above